engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java

```diff
@@ -35,7 +39,7 @@ public static void main(String[] args) throws Exception {
         SortedSetMultimap<String, String> sortedApi = Multimaps.newSortedSetMultimap(new HashMap<>(), TreeSet::new);
 
         for (Class<?> apiClass : environment.getTypesAnnotatedWith(API.class)) {
-            //System.out.println("Processing: " + apiClass);
+            LOGGER.debug("Processing: {}", apiClass);
```

Comment:
This is something that is concerning me a bit. The class defines its own `main` method, which means that it does not use the usual entrypoint. It looks like a utility program to me. Often the printed outputs of utility programs are parsed by other tools, so it may have been intentional in this rare case to not use a logger framework.

The logger framework usually prepends all outputs lines with the log level and class, which would potentially break programs that depended on the existing output format. I do not know where this tool is used, so I am unable to say definitively if this is the case.

Reply:
hmm. you are right @BenjaminAmos . we try to get docmentation via these, i found : https://github.com/MovingBlocks/Terasology/issues/1975 ? @Cervator ?

Reply:
this occurence is for debugging, but others are not. reverted.

Reply:
These changes still seem to be in... @soloturn you might want to recheck your revert

Reply:
no, they are not in any more. the comment was on a line which is part of logging.

Reply:
maybe I misunderstood, but I thought @BenjaminAmos's point was that we shouldn't use the logger in `main` here but stick with `System.out.println`, right?
however, I still see the logger field and logger usages in this file...
@soloturn can you clarify why you say it has been reverted when it looks like it hasn't been? ö.Ö

Reply:
it outputs documentation, below run ApiScraper from branch "develop". things are commented to give additional hints, but messages like "processing blabla " is clearly not part of documentation, therefor commented, now it is not commented but is standard logging which can be switched on and off without changing the code.

```
00:39:15.601 [main] DEBUG org.terasology.gestalt.module.sandbox.PermissionSet -- Globally granted permission '("java.util.PropertyPermission" "reactor.schedulers.defaultBoundedElasticQueueSize" "read")'
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.terasology.engine.core.module.ModuleManager (file:/srv/http/repos/terasology/engine/build/classes/)
WARNING: Please consider reporting this to the maintainers of org.terasology.engine.core.module.ModuleManager
WARNING: System::setSecurityManager will be removed in a future release
# Modding API:

## terasology engine

* org.terasology.engine.audio (PACKAGE)
* org.terasology.engine.audio.events (PACKAGE)
* org.terasology.engine.config.Config (CLASS)
* org.terasology.engine.config.RenderingConfig (CLASS)
* org.terasology.engine.config.RenderingDebugConfig (CLASS)
* org.terasology.engine.config.facade.TelemetryConfiguration (INTERFACE)
* org.terasology.engine.config.flexible.Setting (INTERFACE)
* org.terasology.engine.config.flexible.constraints (PACKAGE)
* org.terasology.engine.context.Context (INTERFACE)
* org.terasology.engine.core.ComponentFieldUri (CLASS)
* org.terasology.engine.core.ComponentSystemManager (CLASS)
* org.terasology.engine.core.GameScheduler (CLASS)
* org.terasology.engine.core.Observer (INTERFACE)
* org.terasology.engine.core.SimpleUri (CLASS)
* org.terasology.engine.core.Time (INTERFACE)
* org.terasology.engine.core.Uri (INTERFACE)
* org.terasology.engine.core.bootstrap.ClassMetaLibrary (INTERFACE)
* org.terasology.engine.core.modes.loadProcesses.AwaitedLocalCharacterSpawnEvent (CLASS)
* org.terasology.engine.core.module.ModuleInputStream (CLASS)
* org.terasology.engine.core.module.ModuleOutputStream (CLASS)
* org.terasology.engine.core.module.SandboxFileManager (CLASS)
* org.terasology.engine.core.module.rendering (PACKAGE)

and so on and so on ...
```

Reply:
> this is now println again. other things are commented [...] now it is not commented but is standard logging

I'm so sorry, this is probably my fault for not understanding, but I still think we're not talking about the same thing.

As @BenjaminAmos said: All of the parts prepended to the message (by using output of a logger like ``Logger.info()`` instead of without a logger like ``System.out.println()``) might upset programs which were expecting just a plain output format.

Based on this, I assume we want to use ``System.out.println`` (no logger) to avoid potential issues with the parts prepended by using a logger. Do you concur with that @soloturn ?

If you don't, then I'd like to understand, why not.

If you do, I don't understand why there's still so many logger uses in the current state of the PR, see example screenshots below:
![grafik](https://github.com/MovingBlocks/Terasology/assets/29981695/e03f635e-ce65-4cd2-9029-faf616e3cfaf)
![grafik](https://github.com/MovingBlocks/Terasology/assets/29981695/b4541220-e42e-48c5-915b-0783d875e23f)

Reply:
you most likely executed the main method, isn't it? if you could tell me where you expect println and not a logger i'd be glad. take for example the "unknown protocol for .." message, this is not something which should go into a documentation, its just a bug or warning to be fixed.

Reply:
I didn't execute anything (yet), I only looked at the files changed and noticed more logger usages than I expected (see screenshots in my previous comment.